### PR TITLE
Add sample JavaScript files

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
NOTE: this is very WIP, it's been a long while since I've done any ruby.

The tldr (and TODO list):
 - [ ] add hbs template-literal syntax highlighting (glimmer) to both javascript and typescript (sub language highlighting)
 - [ ] add `<template>` syntax higlighting (glimmer) to both javascript and typescript (sub language highlighting)
 
 Examples of the syntax:
  - Highlighting
    - [Highlight.JS](hljs-glimmer.nullvoxpopuli.com/)
    - [Prism.JS](https://prismjs-glimmer.nullvoxpopuli.com/)
    - [TreeSitter](https://github.com/alexlafroscia/tree-sitter-glimmer)
    - [VSCode](https://marketplace.visualstudio.com/items?itemName=chiragpat.vscode-glimmer)
  - [library + babel plugin](https://github.com/ember-template-imports/ember-template-imports)
  - [REPL](https://glimmerjs.github.io/glimmer-experimental/)
  - [REPL (embedded markdown)](https://limber.glimdown.com/?t=%23%20Limber%20Editor%0A%0A%23%23%20With%20JavaScript%0A%0Acode%20fence%20tags%20can%20be%20used%20to%20directly%20render%20components%2C%0Aas%20well%20as%20render%20the%20code%20snippet%20the%20live%20code%20comes%20from.%0A%0AOptions%3A%0A%0A%20-%20%60live%60%20-%20_renders%20in%20place_%0A%20-%20%60live%20preview%60%20_renders%20in%20place%2C%20placing%20the%20source%20after%20the%20live%20render_%0A%20-%20%60live%20preview%20below%60%20_renders%20in%20place%2C%20placing%20the%20live%20render%20below%20the%20source_%0A%0A%60%60%60gjs%20live%0A%3Ctemplate%3E%0A%20%20Hello%2C%20Glimmer!%0A%3C%2Ftemplate%3E%0A%60%60%60%0A%0A%60%60%60gjs%20live%20preview%0Aimport%20Component%20from%20%27%40glimmer%2Fcomponent%27%3B%0Aimport%20%7B%20tracked%20%7D%20from%20%27%40glimmer%2Ftracking%27%3B%0Aimport%20%7B%20on%20%7D%20from%20%27%40ember%2Fmodifier%27%3B%0A%0Aexport%20default%20class%20HelloWorld%20extends%20Component%20%7B%0A%20%20%40tracked%20count%20%3D%200%3B%0A%0A%20%20increment%20%3D%20()%20%3D%3E%20this.count%20%2B%3D%201%3B%0A%0A%20%20%3Ctemplate%3E%0A%20%20%20%20%3Cp%3EYou%20have%20clicked%20the%20button%20%7B%7Bthis.count%7D%7D%20times.%3C%2Fp%3E%0A%0A%20%20%20%20%3Cbutton%20%7B%7Bon%20%22click%22%20this.increment%7D%7D%3EClick%3C%2Fbutton%3E%0A%20%20%3C%2Ftemplate%3E%0A%7D%0A%60%60%60%0A)
   

## Description
Ember is introducing a new syntax to help improve developer experience and more easily onboard new developers.

In addition, this PR will also add support for the `hbs` template literal, which is a current syntax and api used throughout ember for many many years, and is becoming more prominent as intermediary template-only components are written while mainstream folks wait for `<template>` support.

## Checklist:

- [ ] **I am associating a language with a new file extension.**
  - [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s): 
      - [URL to each sample source, if applicable]: 
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am adding a new language.**
  - [ ] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  
  - [ ] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

- [x] **I am fixing a misclassified language**
  - [x] I have included a new sample for the misclassified language:
    - Sample source(s): 
      - [URL to each sample source, if applicable]: https://github.com/NullVoxPopuli/highlightjs-glimmer
    - Sample license(s): MIT
  - [x] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  
  - Old: https://github-lightshow.herokuapp.com/
  - New: https://github-lightshow.herokuapp.com/

- [ ] **I am updating a grammar submodule**
  

- [x] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

- [ ] **I am changing the color associated with a language**
  
  - [ ] I have obtained agreement from the wider language community on this color change.
    - [URL to public discussion]
    - [Optional: URL to official branding guidelines for the language]

### CODE REVIEW

**Comment by Alhadis** [comment]
Improving syntax highlighting is less to do with Linguist and much more to do with the [individual repositories](https://github.com/github/linguist/blob/master/vendor/README.md) Linguist uses to provide highlighting.

In other words, I recommend preparing a submission to [`tree-sitter/tree-sitter-javascript`](https://github.com/tree-sitter/tree-sitter-javascript) and/or [`tree-sitter/tree-sitter-typescript`](https://github.com/tree-sitter/tree-sitter-typescript).

**Comment by NullVoxPopuli** [comment]
@Alhadis I had originally considered that, but tree-sitter-javascript and typescript have already been updated, yet the highlighting on github does not reflect the same highlighting. 

For example: 
hbs template literal: https://github.com/NullVoxPopuli/limber/blob/419c4d80878dbad64536d3e047ae6d70a924b486/frontend/app/components/shadowed.ts#L17
image
yet in my editor, which uses treesitter, I see: 
image

Is it possible that tree-sitter-javascript and tree-sitter-typescript in this repo arer _very_ out of date?

**Comment by NullVoxPopuli** [comment]
:thinking: hmm, the version seems up to date :thinking:

**Comment by NullVoxPopuli** [comment]
how do I preview syntax highlighting? the CONTRIBUTING.md didn't say how :thinking:

**Comment by Alhadis** [comment]
> how do I preview syntax highlighting? the CONTRIBUTING.md didn't say how 🤔

Uh, [yes it does](https://github.com/github/linguist/blob/HEAD/CONTRIBUTING.md#fixing-syntax-highlighting):

> You can test grammars using [Lightshow](https://github-lightshow.herokuapp.com/).

As for this…

> Is it possible that `tree-sitter-javascript` and `tree-sitter-typescript` in this repo arer very out of date?

… the answer is at the top of the [grammar index](https://github.com/github/linguist/tree/e2f638d17598e84f6f91c2d03c2f3e27f22a2163/vendor#grammar-index):

> **Note:** grammars marked with 🐌 are not updated when Linguist is so upstream fixes may take longer to appear on GitHub.

**Comment by NullVoxPopuli** [comment]
> Uh, yes it does:

ok, but how do I link the project locally?

**Comment by Alhadis** [comment]
No, there isn't. Moreover, [Glimmer](https://github.com/chiragpat/vscode-glimmer/blob/master/syntaxes/inline-hbs.json) appears to be ordinary JavaScript/TypeScript, so unless it has a distinguishing file extension (say, `foo.glimmer` instead of `foo.js`), this topic is out-of-scope for this repository.

If there *is* a distinguishing file-extension, and it satisfies [the 200+ users criteria](https://github.com/github/linguist/blob/HEAD/CONTRIBUTING.md#adding-a-language), then Glimmer can be added as a new language entry.

**Comment by NullVoxPopuli** [comment]
So you're saying nested languages are never going to be supported by github?
I see:
  gql is lightly supported: https://github.com/apollographql/graphql-tag
  html is very well supported: https://github.com/lit/lit/tree/main/packages/lit-html
  jsx is on top of javascript as well (rather than a separate file ext: https://gist.github.com/NullVoxPopuli/039f66a516c4ed4ede2088daa91f5d5a
  
adding support for glimmer seems totally in-scope for what is already supported

**Comment by Alhadis** [comment]
> * gql is lightly supported: https://github.com/apollographql/graphql-tag
> * html is very well supported: https://github.com/lit/lit/tree/main/packages/lit-html
> * jsx is on top of javascript as well (rather than a separate file ext: https://gist.github.com/NullVoxPopuli/039f66a516c4ed4ede2088daa91f5d5a

Yes, each of those formats has a dedicated injection rule defined by a Tree-Sitter grammar, somewhere. We also support [GraphQL](https://github.com/github/linguist/blob/e2f638d17598e84f6f91c2d03c2f3e27f22a2163/lib/linguist/languages.yml#L2185-L2194), [HTML](https://github.com/github/linguist/blob/e2f638d17598e84f6f91c2d03c2f3e27f22a2163/lib/linguist/languages.yml#L2273-L2290), and [Handlebars](https://github.com/github/linguist/blob/e2f638d17598e84f6f91c2d03c2f3e27f22a2163/lib/linguist/languages.yml#L2403-L2414). Each of those languages meets the usage criteria I mentioned above.

> my current hunch as to why embedded `hbs` and `<template>` currently don't work on github is because we need tree-sitter-glimmer installed to enable the nesting

That's correct.

**Comment by NullVoxPopuli** [comment]
closing in favor of: https://github.com/github/linguist/pull/5632

**Comment by lildude** [comment]
Regarding…

> @Alhadis I had originally considered that, but tree-sitter-javascript and typescript have already been updated, yet the highlighting on github does not reflect the same highlighting.

… and …

> Is it possible that tree-sitter-javascript and tree-sitter-typescript in this repo arer very out of date?

It's quite possible, as stated in the `vendor/README.md` file:

> Note: grammars marked with 🐌 are not updated when Linguist is so upstream fixes may take longer to appear on GitHub.
